src/main/java/juja/sqlcmd/DatabaseManager.java

```diff
@@ -5,16 +5,33 @@
 import java.sql.SQLException;
 
 public class DatabaseManager {
+    private final String jdbcDriverClass;
+    private final String jdbcUrl;
     private Connection connection;
 
+    public DatabaseManager() {
+        jdbcDriverClass = "org.postgresql.Driver";
+        jdbcUrl = "jdbc:postgresql://localhost:5432/";
+    }
+
+    public DatabaseManager(String jdbcDriverClass, String jdbcUrl) {
+        this.jdbcDriverClass = jdbcDriverClass;
+        this.jdbcUrl = jdbcUrl;
+    }
+
     boolean connect(String database, String user, String password) {
         try {
-            Class.forName("org.postgresql.Driver");
+            Class.forName(jdbcDriverClass);
+            if (connection != null) {
+                connection.close();
```

Comment:
Maybe it's better to notify the client somehow that already there is unclosed connection? That's the moment when you provide additional logic to your implementation without requirements in a task. (http://bit.ly/2jRFHEn)

Reply:
I just copied it from Baglai's example but actually despite these lines look reasonable, I don't see any hanging connections in db without them. May be it's redundant